stan/math/prim/fun/inv_Phi.hpp

```diff
@@ -37,45 +40,112 @@ inline double inv_Phi(double p) {
+  static const double f[7]
+      = {5.99832206555887937690e-01, 1.36929880922735805310e-01,
+         1.48753612908506148525e-02, 7.86869131145613259100e-04,
+         1.84631831751005468180e-05, 1.42151175831644588870e-07,
+         2.04426310338993978564e-15};
 
-  static const double p_low = 0.02425;
-  static const double p_high = 0.97575;
+  double q = p - 0.5;
+  double r;
+  double val;
 
-  double x;
-  if ((p_low <= p) && (p <= p_high)) {
-    double q = p - 0.5;
-    double r = square(q);
-    x = (((((a[0] * r + a[1]) * r + a[2]) * r + a[3]) * r + a[4]) * r + a[5])
-        * q
-        / (((((b[0] * r + b[1]) * r + b[2]) * r + b[3]) * r + b[4]) * r + 1.0);
-  } else if (p < p_low) {
-    double q = std::sqrt(-2.0 * std::log(p));
-    x = (((((c[0] * q + c[1]) * q + c[2]) * q + c[3]) * q + c[4]) * q + c[5])
-        / ((((d[0] * q + d[1]) * q + d[2]) * q + d[3]) * q + 1.0);
+  if (std::fabs(q) <= .425) {
+    r = .180625 - square(q);
+    return q
+           * (((((((a[7] * r + a[6]) * r + a[5]) * r + a[4]) * r + a[3]) * r
+                + a[2])
+                   * r
+               + a[1])
+                  * r
+              + a[0])
+           / (((((((b[6] * r + b[5]) * r + b[4]) * r + b[3]) * r + b[2]) * r
+                + b[1])
+                   * r
+               + b[0])
+                  * r
+              + 1.0);
   } else {
-    double q = std::sqrt(-2.0 * log1m(p));
-    x = -(((((c[0] * q + c[1]) * q + c[2]) * q + c[3]) * q + c[4]) * q + c[5])
-        / ((((d[0] * q + d[1]) * q + d[2]) * q + d[3]) * q + 1.0);
-  }
+    r = q < 0 ? p : 1 - p;
+
+    if (r <= 0)
+      return 0;
 
-  if (x < 37.6) {  // gradient blows up past here
-    double e = Phi(x) - p;
-    double u = e * SQRT_TWO_PI * std::exp(0.5 * square(x));
-    x -= u / (1.0 + 0.5 * x * u);
+    r = std::sqrt(-std::log(r));
+
+    if (r <= 5.0) {
+      r += -1.6;
+      val = (((((((c[7] * r + c[6]) * r + c[5]) * r + c[4]) * r + c[3]) * r
+               + c[2])
+                  * r
+              + c[1])
+                 * r
+             + c[0])
+            / (((((((d[6] * r + d[5]) * r + d[4]) * r + d[3]) * r + d[2]) * r
+                 + d[1])
+                    * r
+                + d[0])
+                   * r
+               + 1.0);
+    } else {
+      r -= 5.0;
+      val = (((((((e[7] * r + e[6]) * r + e[5]) * r + e[4]) * r + e[3]) * r
+               + e[2])
+                  * r
+              + e[1])
+                 * r
+             + e[0])
+            / (((((((f[6] * r + f[5]) * r + f[4]) * r + f[3]) * r + f[2]) * r
+                 + f[1])
+                    * r
+                + f[0])
+                   * r
+               + 1.0);
+    }
+    if (q < 0.0)
+      return -val;
   }
+  return val;
+}
+}  // namespace internal
 
-  return x;
+/**
+ * The inverse of the unit normal cumulative distribution function.
+ *
+ * The ternary operation is to protect against floating point error
```

Comment:
The implementation details don't belong in the doc.  It would be OK to talk about precision.